tools/util.py

```diff
@@ -374,14 +376,12 @@ def mkdtemp():
 
 
 class DenoTestCase(unittest.TestCase):
-    @property
-    def build_dir(self):
-        args = test_args()
-        return args.build_dir
+    @classmethod
+    def setUpClass(cls):
+        args = parse_test_args()
```

Comment:
This will be run for each TestCase? We should cache (run once) as you suggested.

Reply:
Yep, I'll rewrite it as metaclass to cache it once

Reply:
I tried with metaclass, it will still be executed multiple times just like `setUpClass` - IMHO current solution is sufficient and Pythonic, so let's leave it that way?